api-put-object-multipart.go

```diff
@@ -114,7 +117,15 @@ func (c Client) putObjectMultipartStream(bucketName, objectName string, reader i
 	for partNumber <= totalPartsCount {
 		// Calculates MD5 and SHA256 sum while copying partSize bytes
 		// into tmpBuffer.
-		md5Sum, sha256Sum, prtSize, rErr := c.hashCopyN(tmpBuffer, reader, partSize)
+		hashSums := make(map[string][]byte)
+		hashAlgos := make(map[string]hash.Hash)
+
+		hashAlgos["md5"] = md5.New()
+		if c.signature.isV4() && !c.secure {
+			hashAlgos["sha256"] = sha256.New()
+		}
+
+		prtSize, rErr := hashCopyN(hashAlgos, hashSums, tmpBuffer, reader, partSize)
```

Comment:
with this approach, hashCopyN and related functions (hashCopy hashCopyBuffer) can be eliminated by doing:

```
md5Writer := md5.New()
writers := []io.Writers{tmpBuffer, md5Writer}

var sha256Writer hash.Hash

if c.signature.isV4() && !c.secure {
    sha256Writer = sha256.New()
    writers = append(writers, sha256Writer)
}

wn, err := io.CopyN(io.MultiWriter(writers...), reader, partSize)
```